Common/src/geometry_structure.cpp

```diff
@@ -9386,1429 +9383,1580 @@ void CPhysicalGeometry::Read_CGNS_Format_Parallel(CConfig *config, string val_me
       
-      gridCoords[j-1] = new passivedouble*[ncoords];
-      for (int ii = 0; ii < ncoords; ii++) {
-        *(gridCoords[j-1]+ii) = new passivedouble[nPoint];
+      if (rank == MASTER_NODE) {
+        cout << "Loading volume section " << string(sectionNames[s].data());
+        cout <<  " from file." << endl;
       }
+
+      /*--- Find the number of nodes required to represent
+       this type of element. ---*/
       
-      /*--- Loop over each set of coordinates. Note again
-       that the indexing starts at 1. ---*/
+      ElementType_t elmt_type;
+      if (cg_npe(elemType, &npe)) cg_error_exit();
       
-      for (int k = 1; k <= ncoords; k++) {
+      /*--- Loop through all of the elements in this section to get more
+       information and to decide whether it has internal elements. ---*/
+
+      if (elemType == MIXED) {
+        isMixed = true;
+      }
+
+      unsigned long counter = 0;
+      for (iElem = 0; iElem < nElems[s]; iElem++) {
         
-        /*--- Read the coordinate info. This will retrieve the
-         data type (either RealSingle or RealDouble) as
-         well as the coordname which will specifiy the
-         type of data that it is based in the SIDS convention.
-         This might be "CoordinateX," for instance. ---*/
+        /*--- If we have a mixed element section, we need to check the elem
+         type one by one. Set the flag to true if mixed. ---*/
         
-        if (cg_coord_info(fn, i, j, k, &datatype, coordname))
-          cg_error_exit();
-        if (rank == MASTER_NODE) {
-          cout << "Loading " << coordname;
-          if (size > SINGLE_NODE) {
-            cout << " values into linear partitions." << endl;
-          } else {
-            cout << " values." << endl;
-          }
+        if (isMixed) {
+          elmt_type = ElementType_t(connElemCGNS[counter]);
+          npe = connOffsetCGNS[iElem+1]-connOffsetCGNS[iElem]-1;
+          counter++; for (int jj = 0; jj < npe; jj++) counter++;
+        } else {
+          elmt_type = elemType;
         }
         
-        /*--- Always retrieve the grid coords in su2double precision. ---*/
+        /*--- Store the number of verts per elem for the current elem. ---*/
+        
+        nPoinPerElem[iElem] = npe;
+        
+        /*--- Store the global ID for this element. Note the -1 to move
+         from CGNS convention to SU2 convention. We also subtract off
+         an additional offset in case we have found boundary sections
+         prior to this one, in order to keep the internal element global
+         IDs indexed starting from zero. ---*/
         
-        if (datatype != RealDouble) {
-          SU2_MPI::Error("CGNS coordinates are not double precision.", CURRENT_FUNCTION);
+        elemGlobalID[iElem] = elemB[rank] + iElem - 1 - elemOffset[s];
+        
+        switch (elmt_type) {
+          case NODE:
+            currentElem   = "Vertex";
+            elemTypes[iElem] = 1;
```

Comment:
I think I saw this kind of information already as I was scrolling down, is there a case to encapsulate this in some `getName`, `getCode` type methods?